### ISSUE TITLE
Enable network container options for Flannel (k3s)

### ISSUE THREAD

**Issue by slickwarren** [rancher/dashboard#7005]
**Is your feature request related to a problem? Please describe.**
an option for k3s is flannel. We should allow the user to select from a list of available options (as we do for rke1/2)

**Describe the solution you'd like**
* allow user to select flannel
* support the following flags: https://rancher.com/docs/k3s/latest/en/installation/network-options/#flannel-options

**Describe alternatives you've considered**
n/a

**Additional context**
this will be required for k3s GA
supporting backend ticket: https://github.com/rancher/rancher/issues/39128

**Comment by manuelbuil** [comment]
I agree, we should allow the user select the `flannel-backend`. Note that you are pointing at an outdated docs link and there is no `--flannel-backend=wireguard` anymore. Now it is `--flannel-backend=wireguard-native`.

**Comment by catherineluse** [comment]
> this will be required for k3s GA

Although K3s cluster provisioning will be GA as of 2.7.0, this ticket is scheduled for 2.7.2. @gaktive can you confirm it isn't required for K3s provisioning GA?

**Comment by gaktive** [comment]
Present related work out of milestones. Pushing to Q4 just so I can keep an eye on this later.
